src/systems/systems/src/air_conditioning/mod.rs

```diff
@@ -244,14 +251,47 @@ impl PackFlowValve {
         &mut self,
         context: &UpdateContext,
         open_fcv: &impl ControllerSignal<PackFlowValveSignal>,
+        engines: [&impl EngineCorrectedN1; 2],
+        pneumatic: &(impl PneumaticBleed + EngineStartState),
+        pneumatic_overhead: &impl EngineBleedPushbutton,
     ) {
-        if let Some(signal) = open_fcv.signal() {
-            self.is_open = signal.target_open_amount(self.number) > Ratio::new::<percent>(0.)
+        if self.can_move_fcv(engines, pneumatic, pneumatic_overhead) {
+            if let Some(signal) = open_fcv.signal() {
+                self.is_open = signal.target_open_amount(self.number) > Ratio::new::<percent>(0.)
+            }
+            if self.is_open {
+                self.timer_open += context.delta();
+            } else {
+                self.timer_open = Duration::from_secs(0);
+            }
+        } else {
+            self.is_open = false;
         }
-        if self.is_open {
-            self.timer_open += context.delta();
+    }
+
+    fn can_move_fcv(
+        &self,
+        engines: [&impl EngineCorrectedN1; 2],
+        pneumatic: &(impl PneumaticBleed + EngineStartState),
+        pneumatic_overhead: &impl EngineBleedPushbutton,
+    ) -> bool {
+        // Pneumatic overhead represents engine bleed pushbutton for left and right engine(s)
+        if self.number == 1 {
+            ((engines[0].corrected_n1() >= Ratio::new::<percent>(15.)
+                && pneumatic_overhead.left_engine_bleed_pushbutton_is_auto())
+                || (engines[1].corrected_n1() >= Ratio::new::<percent>(15.)
+                    && pneumatic_overhead.right_engine_bleed_pushbutton_is_auto()
+                    && pneumatic.engine_crossbleed_is_on()))
+                || pneumatic.apu_bleed_is_on()
+        } else if self.number == 2 {
+            ((engines[1].corrected_n1() >= Ratio::new::<percent>(15.)
+                && pneumatic_overhead.right_engine_bleed_pushbutton_is_auto())
+                || (engines[0].corrected_n1() >= Ratio::new::<percent>(15.)
+                    && pneumatic_overhead.left_engine_bleed_pushbutton_is_auto()
+                    && pneumatic.engine_crossbleed_is_on()))
+                || pneumatic.apu_bleed_is_on()
```

Comment:
**thought**: A lot of duplication in this function originates from having methods which don't work on numbers/indexes, but are named `left` or `right`.
**suggestion**: Do consider changing methods such that they accept an engine number or index, as you'd then be able to remove this duplication.

Reply:
You are right, I've changed this now to reduce duplication. It will need to be refactored in the future to allow for 4 engines, but I'll leave that for a future PR.

Reply:
**thought**: I find this code somewhat difficult to understand. I only understood it fully by considering both cases:

If `number == 2`, then this becomes:
```rust
        ((engines[1].corrected_n1() >= Ratio::new::<percent>(15.)
            && pneumatic_overhead.engine_bleed_pushbuttons_are_auto()[1])
            || (engines[0].corrected_n1() >= Ratio::new::<percent>(15.)
                && pneumatic_overhead.engine_bleed_pushbuttons_are_auto()
                    [0]
                && pneumatic.engine_crossbleed_is_on()))
            || pneumatic.apu_bleed_is_on()
```
If `number == 1`, then this becomes:
```rust
        ((engines[0].corrected_n1() >= Ratio::new::<percent>(15.)
            && pneumatic_overhead.engine_bleed_pushbuttons_are_auto()[0])
            || (engines[1].corrected_n1() >= Ratio::new::<percent>(15.)
                && pneumatic_overhead.engine_bleed_pushbuttons_are_auto()
                    [1]
                && pneumatic.engine_crossbleed_is_on()))
            || pneumatic.apu_bleed_is_on()
```
Do consider working on improving this (and making the design work work for any number of engines, as this is in the `systems` crate, not `a320_systems`). If you want to do so in a separate PR that is of course fine 😀.